.changeset/forty-crabs-judge.md

```diff
@@ -0,0 +1,30 @@
+---
+'astro': patch
+---
+
+Adds an optional middleware for usage with `astro:env`
+
+If you're using `astro:env`, you can now use a middleware to detect server envrionment variables leaks on the client:
+
+```ts
+import { leakDetectionMiddleware } from 'astro/env/middleware'
```

Comment:
I wonder if we should make this `astro:env/middleware` instead?

Reply:
I think it makes things a bit more complicated. There's no codegen involved for the middleware code so I'm a bit reluctant to move it to a virtual import. We were also doing this for `astro:env/setup` and ended up moving it to `astro/env/setup`